__tests__/pet.test.js

```diff
@@ -0,0 +1,125 @@
+    it('increases the fitness by +4', () => {
+          const petOne = new Pet('Fido');
+          petOne.fitness = 4;
+          petOne.walk();
+          expect(petOne.fitness).toEqual(8);
+    }); 
+});
+describe('feed', () => {
+        it('decreases the hunger by 3', () => {
+              const petOne = new Pet('Fido');
+              petOne.hunger = 6;
+              petOne.feed();
+              expect(petOne.hunger).toEqual(3);
+        });     
+}); 
+describe('walk', () => {
+    it('increases the fitness to a MAX of 10', () => {
+        const petOne = new Pet('Fido');
+        petOne.fitness = 8;
+        petOne.walk();
+        expect(petOne.fitness).toEqual(10);
+    }); 
+});
+describe('constructor', () => {
+    it('has an initial hunger of 0', () => {
+      const petOne = new Pet('Fido');
+      expect(petOne.hunger).toEqual(0);
+    });
+});
+describe('constructor', () => {
+    it('has an initial fitness of 10', () => {
+      const petOne = new Pet('Fido');
+      expect(petOne.fitness).toEqual(10);
+    });
+});
+describe('checkUp status', () => {
+    it('checks the health of the pet', () => {
+        const petOne = new Pet('Fido');
+        petOne.fitness = 0;
+        petOne.hunger = 1;
+        expect(petOne.fitness <= 3 && petOne.hunger <=5).toBe(true);
+        
+        petOne.fitness = 4;
+        petOne.hunger = 6;
+        expect(petOne.hunger >= 5 && petOne.fitness >=3).toBe(true);
+        
+        petOne.fitness = 2;
+        petOne.hunger = 6;
+        expect(petOne.fitness <= 3 && petOne.hunger >=5).toBe(true);
+       
+        petOne.fitness = 4;
+        petOne.hunger = 4;
+        expect(petOne.fitness >= 3 && petOne.hunger <=5).toBe(true);
+    }); 
+});
+describe('feed', () => {
+    it('throws an error if the pet is not alive', () => {
+        const petOne = new Pet('Fido');
+        petOne.age = 30;
+        expect(() => petOne.feed()).toThrow('Your pet is no longer alive!!');
+    });
+});
+describe('walk', () => {
+    it('throws an error if the pet is not alive', () => {
+        const petOne = new Pet('Fido');
+        petOne.fitness = -1;
+        expect(() => petOne.feed()).toThrow('Your pet is no longer alive!!');
+    });
+});
+describe('growUp', () => {
+    it('throws an error if the pet is not alive', () => {
+        const petOne = new Pet('Fido');
+        petOne.age = 30;
+        expect(() => petOne.feed()).toThrow('Your pet is no longer alive!!');
+    });
+});
+describe('Child Adoption Success', () => {
+    it('throws an error if the child is not alive', () => {
+        const child = new Pet('Amelia');
+        child.age = 30;
+        expect(() => child.feed()).toThrow('Your pet is no longer alive!!');
+    });
+});
+describe('Baby Born Success', () => {
+    it('throws an error if the child is not alive', () => {
+        const baby = new Pet('Billy');
+        baby.age = 30;
+        expect(() => baby.feed()).toThrow('Your pet is no longer alive!!');
+    });
+});
```

Comment:
This should be looked at a bit more closely. The tests dont actually test any of the child methods like `adoptChild` or `haveBaby`, plus they are both currently testing the exact same functionality. You could wrap a few tests covering similar functions together in a single describe block, something like:
```
describe("Children methods", () => {
    it("parent can have a baby", () => {
        const parent = new Pet("Bob");
        const baby = new Pet("Billy");
        parent.haveBaby(baby)

        expect(parent.children).toEqual([baby]);
    });
  
    it("parent can adopt a baby", () => {
        const parent = new Pet("Bob");
        const baby = new Pet("Billy");
        parent.adoptChild(baby)

        expect(parent.children).toEqual([baby]);
    });
  
    it("throws an error if the child is not alive", () => {
        const parent = new Pet("Bob");
        const child = new Pet("Amelia");
        parent.adoptChild(child)
        parent.children[0].age = 30;
  
        expect(() => child.feed()).toThrow("Your pet is no longer alive!!");
    });
});
```
This actually tests the functions they should and is a bit more cleanly organised